### ISSUE TITLE
Array arguments

### ISSUE THREAD

**Issue by idelvall** [earthly/earthly#3435]
It would be useful to have a way to pass an array as an argument, without requiring the developer to deal with its serialization/deserialization.

In particular, I would like that repeated args are available as a bash array:

```earthly
foo:
  DO +BAR \
    --color=red
    --color=blue
    --color=green

BAR:
  COMMAND
  ARG color
  RUN \
    for c in "${color[@]}"; \
      do echo $c; \
    done
``` 

Right now, arg repetition makes the target being called multiple times in parallel, but I found this other behavior more intuitive. 

Wondering what other people think about this

**Comment by vladaionescu** [comment]
Note that the proposed syntax is currently used to represent a matrix target.

This is likely a duplicate of https://github.com/earthly/earthly/issues/1890 (currently on the roadmap). Please reopen if not!

**Comment by idelvall** [comment]
>Note that the proposed syntax is currently used to represent a matrix target.

I know, actually my proposal is to deprecate this behavior in favor of array arguments. Updating the description
